Tests/Single+MoyaSpec.swift

```diff
@@ -0,0 +1,270 @@
+                })
+
+                expect(called).to(beTruthy())
+            }
+
+            it("filters out non-successful status and redirect codes") {
+                let data = Data()
+                let single = Response(statusCode: 404, data: data).asSingle()
+
+                var errored = false
+                _ = single.filterSuccessfulStatusAndRedirectCodes().subscribe { event in
+                    switch event {
+                    case .success(let object):
+                        fail("called on non-success status code: \(object)")
+                    case .error:
+                        errored = true
+                    }
+                }
+
+                expect(errored).to(beTruthy())
+            }
+
+            it("passes through correct status codes") {
+                let data = Data()
+                let single = Response(statusCode: 200, data: data).asSingle()
+
+                var called = false
+                _ = single.filterSuccessfulStatusAndRedirectCodes().subscribe(onSuccess: { _ in
+                    called = true
+                })
+
+                expect(called).to(beTruthy())
+            }
+
+            it("passes through correct redirect codes") {
+                let data = Data()
+                let single = Response(statusCode: 304, data: data).asSingle()
+
+                var called = false
+                _ = single.filterSuccessfulStatusAndRedirectCodes().subscribe(onSuccess: { _ in
+                    called = true
+                })
+
+                expect(called).to(beTruthy())
+            }
+
+            it("knows how to filter individual status code") {
+                let data = Data()
+                let single = Response(statusCode: 42, data: data).asSingle()
+
+                var called = false
+                _ = single.filter(statusCode: 42).subscribe(onSuccess: { _ in
+                    called = true
+                })
+
+                expect(called).to(beTruthy())
+            }
+
+            it("filters out different individual status code") {
+                let data = Data()
+                let single = Response(statusCode: 43, data: data).asSingle()
+
+                var errored = false
+                _ = single.filter(statusCode: 42).subscribe { event in
+                    switch event {
+                    case .success(let object):
+                        fail("called on non-success status code: \(object)")
+                    case .error:
+                        errored = true
+                    }
+                }
+
+                expect(errored).to(beTruthy())
+            }
+        }
+
+        describe("image maping") {
+            it("maps data representing an image to an image") {
+                let image = Image.testPNGImage(named: "testImage")
+                guard let data = image.asJPEGRepresentation(0.75) else {
+                    fail("Failed creating Data from Image")
```

Comment:
probably `fatalError` is better here, as it is not a failed test, but rather an error in preparation (this one is in few more places as well)